### PR TITLE
Scenario cdt obj mort agent spawn

### PR DESCRIPTION
**Pull request by spawnpbbw**
http://51.178.133.139:8080/projects/alliance-v0-dot-4/work_packages/164/activity

### CODE REVIEW

**Comment by spawnpbbw** [comment]
OP#164

File: Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			}
+		}
+
+		private void Reset()
+		{
+			if ( Repeat == true ) 
+			_DeathCount = 0;
+		}
+
+		public override bool Evaluate(ScenarioManager context)
```
Comment (Pierre-HenriLemp) [inline Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs:55]: Maybe a comment on what represent this `Evaluate` method in this context can help to understand what it is doing.

File: Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			if (Character == _DeadAgent.Character.StringId & (Side == SideType.All || (int)victim.Team.Side == (int)Side) )
+			{
+				_DeathCount += 1;
+				_DeadAgent = default;
+			}
+		}
+
+		private void Reset()
+		{
+			if ( Repeat == true )
```
Comment (Pierre-HenriLemp) [inline Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs:51]:
```suggestion
			if ( Repeat ) 
```

File: Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	public class AgentDeathCondition : Condition
+	{
+
+		[ScenarioEditor(label: "Character", tooltip: "ID of the character to check death.")]
+		public string Character;
+		public SideType Side = SideType.All;
+		[ScenarioEditor(label: "Death quota", tooltip: "How many time character need to die to trigger condition.")]
+		public int DeathQuota = 1;
+		[ScenarioEditor(label: "Repeat", tooltip: "If true, it will trigger each time Death quota is reached once again.")]
+		public bool Repeat;
```
Comment (Pierre-HenriLemp) [inline Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs:20]: Convention is to name boolean if "is..."
```suggestion
		public bool isRepeatedOnQuotaReached; // Maybe a better name can be found
```

**Review by Pierre-HenriLemp — COMMENTED** [review]
Here are some smalls suggestions.
This is nice work ! :)

**Review by Pierre-HenriLemp — APPROVED** [review]
chefs-kiss-french-chef

**Pull request merged**